lib/components/viewers/stop-viewer.js

```diff
@@ -285,20 +285,26 @@ class PatternRow extends Component {
       timeFormat
     } = this.props
     // sort stop times by next departure
-    let sortedStopTimes = null
-    if (stopTimes) {
+    let sortedStopTimes = []
+    const hasStopTimes = stopTimes && stopTimes.length > 0
+    if (hasStopTimes) {
       sortedStopTimes = stopTimes.sort((a, b) => {
         const aTime = a.serviceDay + a.realtimeDeparture
         const bTime = b.serviceDay + b.realtimeDeparture
         return aTime - bTime
       })
       // Cap the number of times shown for any Route at 5. TODO: make configurable
       if (sortedStopTimes.length > 0) sortedStopTimes = sortedStopTimes.slice(0, 5)
-      // Do not show any patterns with no departures happening soon.
-      const timeIsOverThreshold = sortedStopTimes[0].realtimeDeparture - getHomeTime(homeTimezone) > ONE_HOUR_IN_SECONDS * 3
-      if (sortedStopTimes[0] && timeIsOverThreshold) {
-        return null
-      }
+      // Do not show any patterns with no departures happening soon or departures
+      // that have already occurred (there appear to be cases where tomorrow's
+      // times will show up when they should not).
+      const secondsUntilFirstTime = sortedStopTimes[0].realtimeDeparture - getHomeTime(homeTimezone)
+      const firstTimeIsOverThreshold = secondsUntilFirstTime > ONE_HOUR_IN_SECONDS * 3
+      const firstTimeIsInThePast = secondsUntilFirstTime < 0
```

Comment:
These two thresholds would be good things to make configurable, but for now it's fine to leave these values hardcoded.